Adds a description for buildpack groups

dmikusa-pivotal opened the pull request:
This is my attempt to document buildpack groups. Thoughts and feedback are appreciated.

Resolves: #147

jkutner commented on content/docs/concepts/components/buildpack-group.md:
```diff
@@ -0,0 +1,53 @@
+
+## Detection with buildpack groups
+
+A builder or meta-buildpack may contain multiple buildpack groups. When the lifecycle executes the detection process, it will process each buildpack group it finds in the order that the groups are specified. For each buildpack group, the lifecycle will execute the detect binary of all buildpacks in that group (these can be executed in parallel) and aggregate the results. The lifecycle will select ...
```
```suggestion
A builder or meta-buildpack may contain multiple buildpack groups. When the lifecycle executes the detection process, it will process each buildpack group it finds in the order that the groups are specified. For each buildpack group, the lifecycle will execute the detect phase of all buildpacks in that group (these can be executed in parallel) and aggregate the results. The lifecycle will select the first builpack group by order where all of the non-optional buildpacks in that group pass detection.
```

jkutner reviewed (changes requested):
one nit

jromero commented:
@dmikusa-pivotal, this looks great to me. Any follow up to the comments?

dmikusa-pivotal commented on content/docs/concepts/components/buildpack-group.md:
```diff
@@ -0,0 +1,53 @@
+
+For example, you might have a buildpack that installs Java and a buildpack that uses Maven to build your application. These two buildpacks can be combined into a group to implement higher-level functionality, specifically that the first one will install Java and the second will use Java to run Maven, which is a Java build tool.
+
+Because you can have many buildpack groups in a builder or meta-buildpack and you can reuse buildpacks, you could have a second buidlpack group that reuses the buildpack to provide Java but uses a third buildpack that provides Gradle to build your application. By doing this, you can create additional high-level functionality without having duplication.
```
I added a link for builders & meta-buildpacks.

dmikusa-pivotal commented:
Sorry, I got busy with some other things. I just addressed the comments.

@samj1912 Do the links I added go far enough to address your concern? 

Thanks for all the feedback!
